### PR TITLE
 ShopcartItemCollection updates to routes and tests

### PR DESCRIPTION
**Pull request by jules203**
# Pull Request Template

## Checklist:

- [x] I have written test cases for this change, and my code passed the test - updated test cases with new api route
- [x] I have attached this PR to an issue/user story

may need some help de-bugging code!

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -337,35 +315,38 @@ def get_shopcart_item(shopcart_id, item_id):
...
-    shopcart_item.add()
-
-    message = shopcart_item.serialize()
-    location_url = url_for("get_shopcart_item",
-                           shopcart_id=shopcart_item.sid, item_id=shopcart_item.id, _external=True)
-    logger.info("ShopcartItem with ID [%s] created.", shopcart_item.id)
-    return make_response(
-        jsonify(message), status.HTTP_201_CREATED, {"Location": location_url}
-    )
+@api.route('/shopcarts/items', strict_slashes=False)
```
Comment (iLtc) [inline service/routes.py:321]: The route should be `/shopcarts/:id/items`

File: service/routes.py [hunk 2]
```diff
@@ -337,35 +315,38 @@ def get_shopcart_item(shopcart_id, item_id):
...
+            shopcart_items = ShopcartItem.find_by_name(args['name'])
+        elif args['available'] is not None:
+            app.logger.info('Filtering by availability: %s', args['available'])
+            shopcart_items = ShopcartItem.find_by_availability(args['available'])
+        else:
+            shopcart_items = ShopcartItem.all()
 
+        results = [shopcart_item.serialize() for shopcart_item in shopcart_items]
+        logger.info('[%s] Shopcart Items returned', len(results))
+        return results, status.HTTP_200_OK
```
Comment (iLtc) [inline service/routes.py:350]: We should also add a POST method here to replace `create_shopcart_items`

File: service/routes.py [hunk 3]
```diff
@@ -337,35 +315,38 @@ def get_shopcart_item(shopcart_id, item_id):
...
-    )
+@api.route('/shopcarts/items', strict_slashes=False)
+class ShopcartItemCollection(Resource):
+    """ Handles all interactions with collections of Shopcart Items """
+    #------------------------------------------------------------------
+    # LIST ALL SHOPCART ITEMS
+    #------------------------------------------------------------------
+    @api.doc('list_shopcart_items')
+    @api.expect(shopcart_item_args, validate=True)
+    @api.marshal_list_with(shopcart_item_model)
```
Comment (iLtc) [inline service/routes.py:329]: Please include `@api.response(200, 'Shopcart Items List')`

**Review by iLtc — CHANGES_REQUESTED** [review]

**Comment by jules203** [comment]
Thank you @iLtc !!

**Review by KHafner — APPROVED** [review]

**Review by nonococoleo — APPROVED** [review]

**Review by iLtc — APPROVED** [review]

**Comment by KHafner** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/b5c2987e-b3d1-4601-9f28-34419832137d?env_id=ibm:yp:us-south "nyu-shopcart-service-f20-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/c869c127-ceda-423d-80b8-f74ee52b7378 "nyu-shopcart-service-f20-pipeline") deployed [shopcarts](https://github.com/NYU-DevOps2020-shopcarts/shopcarts/releases/tag/deploy-ibm-yp-us-south-kfh226-nyu.edu-dev-20201128-183905 "deploy-ibm-yp-us-south-kfh226-nyu.edu-dev-20201128-183905") to [dev](https://console.bluemix.net/apps/4aa42608-55fd-40d6-b27b-d99aa18a0658?env_id=ibm:yp:us-south "ibm:yp:us-south:kfh226@nyu.edu:dev"), including commit a1564bf02f0c1f066ceea9d42a70dcf118400443

**Comment by KHafner** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/b5c2987e-b3d1-4601-9f28-34419832137d?env_id=ibm:yp:us-south "nyu-shopcart-service-f20-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/c869c127-ceda-423d-80b8-f74ee52b7378 "nyu-shopcart-service-f20-pipeline") deployed [shopcarts](https://github.com/NYU-DevOps2020-shopcarts/shopcarts/releases/tag/deploy-ibm-yp-us-south-kfh226-nyu.edu-prod-20201208-144028 "deploy-ibm-yp-us-south-kfh226-nyu.edu-prod-20201208-144028") to [prod](https://console.bluemix.net/apps/ac06a6b3-040d-43d7-b290-6bfbba3f8c41?env_id=ibm:yp:us-south "ibm:yp:us-south:kfh226@nyu.edu:prod"), including commit a1564bf02f0c1f066ceea9d42a70dcf118400443

**Pull request merged**